### PR TITLE
[YUNIKORN-2694] Improve placement rule funtion's test coverage

### PR DESCRIPTION
**Pull request by SP12893678**
### What is this PR for?
Improve placement rule funtion's test coverage
The placement code will now have full coverage.
- placement_test.go
- provided_rule_test.go
- rule_test.go

### What type of PR is it?
* [x] - Test

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2694

### CODE REVIEW

**Comment by wilfred-s** [comment]
Please check the changes in PR #891 as things might clash with this PR.

**Comment by SP12893678** [comment]
> Please check the changes in PR #891 as things might clash with this PR.

Oh, it might happen.
I will rebase the latest commit and retest it after PR #891 is merged.

**Review by manirajv06 — CHANGES_REQUESTED** [review]
#891 has gone in. Can you rebase and resolve conflicts?

File: pkg/scheduler/placement/rule_test.go [hunk 1]
```diff
@@ -115,3 +115,20 @@ func TestReplaceDot(t *testing.T) {
...
+	if name := rule.getName(); name != expected {
+		t.Errorf("expected %s, got %s", expected, name)
+	}
+}
+
+func TestBasicRuleRuleDAO(t *testing.T) {
+	rule := &basicRule{}
+	dao := rule.ruleDAO()
+	expectedName := "unnamed rule"
+	if dao.Name != expectedName {
```
[the remark this answers is not in the document]
Comment (SP12893678) [reply pkg/scheduler/placement/rule_test.go]: OK. I fixed it.

File: pkg/scheduler/placement/rule_test.go [hunk 2]
```diff
@@ -115,3 +115,20 @@ func TestReplaceDot(t *testing.T) {
 		t.Errorf("replace start or end dots failed, name: %s, ", name)
 	}
 }
+
+func TestBasicRuleGetName(t *testing.T) {
+	rule := &basicRule{}
+	expected := "unnamed rule"
+	if name := rule.getName(); name != expected {
```
[the remark this answers is not in the document]
Comment (SP12893678) [reply pkg/scheduler/placement/rule_test.go]: OK. I fixed it.

File: pkg/scheduler/placement/rule_test.go [hunk 3]
```diff
@@ -115,3 +115,20 @@ func TestReplaceDot(t *testing.T) {
 		t.Errorf("replace start or end dots failed, name: %s, ", name)
 	}
 }
+
+func TestBasicRuleGetName(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (SP12893678) [reply pkg/scheduler/placement/rule_test.go]: Great suggestion!

**Comment by SP12893678** [comment]
> #891 has gone in. Can you rebase and resolve conflicts?

Sure! Please take a look.
I also append new test case for new code from testrule.go

**Review by pbacsko — APPROVED** [review]

**Comment by pbacsko** [comment]
+1

**Pull request closed without merging**